pkg/cloudevents/transport/x/doc.go

```diff
@@ -0,0 +1,49 @@
+/*
+
+Package transport/x shows proposed modifications to package transport.
+
+They are here for discussion purposes, anything that we can agree on
+should be merged into the transport package and may involve
+incompatible changes.
+
+Goals
+
+Support for intermediaries that receive and send events via different
+transports, for example adapters, importers, brokers or channels.
+
+Support for reliable messaging qualities of service between different
+transports
+
+Provide the option of blocking Receive vs. callback SetReceiver.
+There are many situations where the blocking style is easier to
+implement and work with.
+
+Overview of changes
+
+Sender/Receiver interfaces that can be implemented and used separately.
```

Comment:
I would like to know how you can see re-implementing send/response for http.

Reply:
I would like response events to go away, see https://github.com/cloudevents/sdk-go/issues/160
Can you comment on that issue with some details of the use-cases for response events? If it is a requirement it will certainly be possible to build it, but IMO it should be an optional extra mode of operation, not presented as the basic API. I would probably do something like:

type RequestMessage interface { Response(Message) }

basically it becomes another QoS alternative where we require a complete Message in response, not just some protocol-specific ack sequence. 

I'd like to  understand the use cases better though - if there is a real requirement for portable, routable response events then the cloudevents spec needs to be updated with a correlationId. The problem of routing responses may be bigger than a single client-server connection.